[0.18] Document Apache Kafka Sink (#3031) (#3059)
knative/docs: docs/eventing/sink/_index.md

---
title: "Sink"
weight: 60
type: "docs"
---

A sink is an Addressable resource that acts as a link

> between the Eventing mesh and an entity or system.

Reviewer: ```suggestion
between the Eventing mesh and event producer or source.
```

> We can connect any source to a sink, such as `PingSource` and `KafkaSink` objects:

Reviewer: ```suggestion
You can connect any type of source to a sink.
The following example shows how you can connect a PingSource to a KafkaSink object:
```

```yaml
apiVersion: sources.knative.dev/v1beta1
kind: PingSource
metadata:
  name: test-ping-source
spec:
  schedule: "*/1 * * * *"
  jsonData: '{"message": "Hello world!"}'
  sink:
    ref:
      apiVersion: eventing.knative.dev/v1alpha1
      kind: KafkaSink
      name: my-kafka-sink
```

> We can connect a `Trigger` object to a sink, so that we can filter events, before sending them to a sink:

Reviewer: ```suggestion
You can also connect a trigger to a sink, so that events can be filtered before they are sent to a sink:
```

```yaml
apiVersion: eventing.knative.dev/v1
kind: Trigger
metadata:
  name: my-service-trigger
spec:
  broker: default
  filter:
    attributes:
      type: dev.knative.foo.bar
      myextension: my-extension-value
  subscriber:
    ref:
      apiVersion: eventing.knative.dev/v1alpha1
      kind: KafkaSink
      name: my-kafka-sink
```

> ## Knative Sinks

Reviewer: ```suggestion
## Knative sinks
```
